src/leveldbwrapper.h

```diff
@@ -168,6 +199,44 @@ class CLevelDBWrapper
     {
         return pdb->NewIterator(iteroptions);
     }
+
+    /**
+     * Return true if the database managed by this class contains no entries.
+     */
+    bool IsEmpty()
+    {
+        leveldb::Iterator* it = NewIterator();
+        it->SeekToFirst();
+        bool empty = !(it->Valid());
+        delete it;
+        return empty;
+    }
+
+    /**
+     * Accessor for obfuscate_key.
+     */
+    std::string ObfuscateKey() const { return obfuscate_key; }
+
+    /**
+     * Return the obfuscate_key as a hex-formatted string.
+     */
+    std::string ObfuscateKeyHex()
```

Comment:
Same here: make it GetObfuscateKeyHex(). Also don't re-implement the hex logic here, but use `HexStr(...)`.